onnxruntime/core/framework/data_transfer.cc

```diff
@@ -0,0 +1,24 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include "core/framework/data_transfer.h"
+
+namespace onnxruntime {
+
+common::Status IDataTransfer::CopyTensor(const Tensor& src, Tensor& dst) const {
+  return CopyTensor(src, dst, 0);
+}
+
+bool CPUDataTransfer::CanCopy(const OrtDevice& src_device, const OrtDevice& dst_Device) const {
+  return src_device.Type() == OrtDevice::CPU && dst_Device.Type() == OrtDevice::CPU;
```

Comment:
Copying from CPU to CPU? I understand that mkldnn does that today, but it's not really required.

Reply:
the goal of this PR is to move the two "CopyTensor" out of execution provider's interfaces. I agree with you that this CPU2CPU copy is not necessary, but I'll keep this right now and remove it later in separate PRs. Sounds good?

Reply:
ok